scorekeeper/src/App.test.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+import { render, cleanup, fireEvent } from 'react-testing-library';
+import 'jest-dom/extend-expect';
+
+describe('The App Component', () => {
+
+  // test for render component
+  it('should render App', () => {
+    render(<App/>);
+    cleanup();
+  })
+
+describe('Display and Dashboard Components', () => {
+  // test for displaying buttons
+  it('should render Dashboard buttons', () => {
+    const { getByText } = render(<App/>);
+    getByText(/strike/i);
+    getByText(/ball/i);
+    getByText(/foul/i);
+    getByText(/hit/i);
+    cleanup();
+  })
+
+  it('should display balls and strikes', () => {
+    const { getByText } = render(<App/>);
+    getByText(/balls/i);
+    getByText(/strikes/i);
+    getByText(/fouls/i);
+    getByText(/hits/i);
+    cleanup();
+  })
+})
+
+describe('Strike Button', () => {
+  // test strike button
+  it('should increment strikes by 1 when strike button is clicked', () => {
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/strike/i);
+    const currentStrikes = getAllByTestId('strikes')[0];
+    fireEvent.click(button);
+    expect(currentStrikes).toHaveTextContent(/Strikes: 1/i)
+    cleanup();
+  })
+
+  // after 3 strikes, everything resets to 0
+  it('should reset all values to 0 after 3 strikes', () => {
+    const { getByText, getAllByTestId } = render(<App />);
+    const button = getByText(/strike/i);
+    const currentStrikes = getAllByTestId('strikes')[0];
+    const currentBalls = getAllByTestId('balls')[0];
+    const currentFouls = getAllByTestId('fouls')[0];
+    const currentHits = getAllByTestId('hits')[0];
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
+    fireEvent.click(button);
```

Comment:
Since you seem to have this same need elsewhere for using `fireEvent.click` on an element multiple times, you could make a helper in your `describe`'s scope:
```javascript
const click = (button, times = 1);

// ... 
// usage in the test above this one:
click(button);

// ...
// usage in this test:
click(button, 4);
```

Reply:
There are also other things which seem to be repeated between tests; for these, I'd look into using [beforeEach and afterEach in the docs here](https://jestjs.io/docs/en/setup-teardown#repeating-setup-for-many-tests).
You could use `beforeEach` for the rendering and `afterEach` for the cleanup at least, I'd think!